flow-typed/box-ui-elements.js

```diff
@@ -542,33 +542,46 @@ type Translations = {
     onTranslate?: Function
 };
 
-type OpenWithIntegrations = {
-    default_app_integration?: AppIntegrationMiniItem,
+type OpenWithAPI = {
+    default_app_integration?: AppIntegrationAPIMiniItem,
     disabled_reasons?: Array<string>,
     icon?: string,
     is_disabled?: boolean,
-    items: Array<OpenWithIntegrationItem>,
+    items: Array<OpenWithAPIItem>,
     should_show_consent_popup?: boolean
 }
 
-type OpenWithIntegrationItem = {
-    app_integration: AppIntegrationMiniItem | AppIntegrationItem,
+type OpenWithAPIItem = {
+    app_integration: AppIntegrationAPIMiniItem,
     disabled_reasons: Array<string>,
     display_order: number,
     icon: string,
     is_disabled: boolean,
     should_show_consent_popup: boolean
 }
 
-type AppIntegrationMiniItem = {
+type AppIntegrationAPIMiniItem = {
     type: 'app_integration',
     id: string
 }
 
-type AppIntegrationItem = {
+type AppIntegrationAPIItem = AppIntegrationAPIMiniItem & {
     name: string,
     description: string
-} & AppIntegrationMiniItem
+} 
+
+type Integration = {
```

Comment:
This flat map should make accessing data easier in the component. With this type we know all of the required fields are present, and we know about default integrations with no additional logic required.